TodoListDaemonWithCert-core/Program.cs

```diff
@@ -117,14 +115,15 @@ private static X509Certificate2 ReadCertificateFromStore(string certName)
             do
             {
                 retry = false;
+                errorCode = 0;
+
                 try
-                {   // ADAL includes an in memory cache, so this call will only send a message to the server if the cached token is expired.
+                {   // ADAL includes an in-memory cache, so this call will only send a message to the server if the cached token is expired.
                     result = await authContext.AcquireTokenAsync(todoListResourceId, certCred);
                 }
-                catch (Exception ex)
+                catch (AdalException ex)
                 {
-                    AdalException exc = ex as AdalException;
-                    if (exc.ErrorCode == "temporarily_unavailable")
```

Comment:
The old code would crash if the exception wasn't an _AdalException_ so might as well just catch _AdalException_.
More generally, _as_ can return null so you always have to test for that possibility. If you are certain of the type of the object then use a hard cast instead.